packages/databases-collections/src/components/collection-fields/collection-fields.jsx

```diff
@@ -0,0 +1,313 @@
+      collection: this.state.fields.collectionName,
+      options: this.buildOptions()
+    });
+  }
+
+  asNumber(value) {
+    return `${value}` ? +value : undefined;
+  }
+
+  buildOptions() {
+    const state = this.state;
+    const cappedOptions = state.isCapped ? {
+      capped: true,
+      size: state.fields.cappedSize ? state.fields.cappedSize : undefined
+    } : {};
+
+    const collationOptions = state.isCustomCollation ? {
+      collation: state.fields.collation
+    } : {};
+
+    const timeSeriesOptions = state.isTimeSeries ? {
+      timeseries: state.fields.timeSeries
+    } : {};
+
+    return {
+      ...collationOptions,
+      ...cappedOptions,
+      ...timeSeriesOptions
+    };
+  }
+
+  renderDatabaseNameField() {
+    return (<FieldSet>
+      <TextInput
+        autoFocus
+        required
+        label="Database Name"
+        onChange={(e) => this.setField('databaseName', e.target.value)}
+      />
+    </FieldSet>);
+  }
+
+  renderCollectionNameField() {
+    return (<FieldSet>
+      <TextInput
+        autoFocus={!this.props.withDatabase}
+        required
+        label="Collection Name"
+        onChange={(e) => this.setField('collectionName', e.target.value)}
+      />
+    </FieldSet>);
+  }
+
+  renderCappedCollectionFields() {
+    return (<CollapsibleFieldset
+      toggled={this.state.isCapped}
+      onToggle={checked => {
+        this.setState({ isCapped: checked }, this.updateOptions);
+      }}
+      label="Capped Collection"
+      helpUrl={HELP_URL_CAPPED}
+      openLink={this.props.openLink}
+      description="Fixed-size collections that support high-throughput operations that insert and retrieve documents based on insertion order."
+    >
+      <TextInput
+        label="size"
+        type="number"
+        description="Maximum size in bytes for the capped collection."
+        onChange={(e) => this.setField('cappedSize', this.asNumber(e.target.value))}
+      />
+    </CollapsibleFieldset>);
+  }
+
+  renderCollationOptions(values) {
+    const unifiedValues = values.map((elem) => ({
+      value: (typeof elem === 'object') ? elem.value : elem,
+      label: (typeof elem === 'object') ? elem.label : elem
+    }));
+    const options = _.sortBy(unifiedValues, 'value');
+
+    return options.map(({value, label}) => {
+      return (<Option
+        key={label}
+        value={`${value}`}
+      >
+        {label}
+      </Option>);
+    });
+  }
+
+  renderCollationFields() {
+    const options = COLLATION_OPTIONS.map((element) => {
+      return (
+        <FieldSet key={element.field}>
+          <Select
+            className={styles['options-select-dropdown']}
+            label={element.field}
+            name={element.field}
+            placeholder={`Select a value${
+              element.required ? '' : ' [optional]'
+            }`}
+            onChange={(val) => this.setCollationField(element.field, val)}
+            usePortal={false}
+            size={Size.Small}
```

Comment:
A bit weird to me that we have TextInput of normal size, but Select of size small, they are noticeably different in size when both are visible:

<img width="226" alt="image" src="https://user-images.githubusercontent.com/5036933/121503066-03973d80-c9e1-11eb-912e-fd3fd13a7bfa.png">

Is that intentional?

Reply:
This was intentional, the two thoughts behind it are the normal sized selects looked a bit large and there are so many collation options it they quickly took up a lot of space. Happy to put them back to normal size, did sync with Claudia on this and she was cool with both if I remember correctly.

Reply:
Ah, okay, this makes sense, thanks for adding some context!